src/workerd/util/duration-alert.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/string.h>
+#include <kj/time.h>
+#include <kj/debug.h>
+
+namespace workerd::util {
+
+class DurationAlert {
+// This is a utility class for instantiating a timer, which will log if it's destructed after a specified time
+// This works by relying on RAII (Scope-Bound Resource Management) to check how much time has elapsed
+// when the object is destructed. This ensures that the time is checked when the timer object goes out of scope
+// thereby timing everything after the initialization of the timer, until the end of the scope where it was
+// instantiated.
+public:
+  DurationAlert(const kj::MonotonicClock& clock, kj::Duration logWarningTime, kj::StringPtr logMessage)
+    : logWarningTime(logWarningTime),
+      logMessage(logMessage),
+      start(clock.now()),
+      clock(clock) {}
```

Comment:
```suggestion
      clock(clock) {}
      
    KJ_DISALLOW_COPY_AND_MOVE(DurationAlert);
```

Reply:
Added this to the PR, but what's the rationale behind this?

Reply:
I suppose having a move constructor for a raii object like this would lead to confusing behavior, as it'd be weird to track what's actually being timed. Copying probably doesn't make sense either for similar reasons, since you'd always want to re-initialize the start time rather than blindly copying.